modules/Bio/EnsEMBL/Utils/IO/GFFSerializer.pm

```diff
@@ -60,25 +64,37 @@ my %strand_conversion = ( '1' => '+', '0' => '.', '-1' => '-');
 =head2 new
 
     Constructor
-    Arg [1]    : Ontology Adaptor
-    Arg [2]    : Optional File handle
-    Arg [3]    : Default source of the features. Defaults to .
-    
+    Arg [1]    : Optional File handle
+    Arg [2]    : Default source of the features. Defaults to .
+
     Returntype : Bio::EnsEMBL::Utils::IO::GFFSerializer
 
+    Warning    : If legacy mandatory 'Ontology Adaptor' first argument provided.
+                 This argument has been removed but method will continue functioning if still provided,
+                 raising only a warning to the user stating that it can be removed.
+
 =cut
 
 sub new {
     my $class = shift;
+    my $arg1 = shift;
+
+# Legacy constructor required an 'Ontology Adaptor' to be passed on as first argument
+# this is to support and keep scripts using the legacy constructor functioning (with an extra warning)
+# Can probably be removed some time in the future
+    if ( check_ref($arg1, "Bio::EnsEMBL::DBSQL::OntologyTermAdaptor" )) {
+        warning("GFF format does not require an instance of Bio::EnsEMBL::DBSQL::OntologyTermAdaptor anymore."
```

Comment:
I think we should use _deprecate()_ from Bio::EnsEMBL::Utils::Exception here instead of _warning()_.

Reply:
changed to a deprecate.
Did not fill in the DEPRECATED.md, as it looks like that is for methods about to be removed...
This deprecation is really just used as a warning to advise users to remove the now unnecessary argument, and full backwards comparability is provided.